codec/seata-codec-kryo/src/main/java/io/seata/codec/kryo/KryoSerializer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.codec.kryo;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.util.Objects;
+
+/**
+ * @author jsbxyyx
+ */
+public class KryoSerializer {
+
+    private final Kryo kryo;
+
+    public KryoSerializer(Kryo kryo) {
+        this.kryo = Objects.requireNonNull(kryo);
+    }
+
+    public Kryo getKryo() {
+        return kryo;
+    }
+
+    public <T> byte[] serialize(T t) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Output output = new Output(baos);
+        kryo.writeClassAndObject(output, t);
+        output.close();
+        return baos.toByteArray();
+    }
+
+    public <T> T deserialize(byte[] bytes) {
+        ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+        Input input = new Input(bais);
+        input.close();
```

Comment:
Close is called before kryo.readClassAndObject(input);Is there a problem?
![image](https://user-images.githubusercontent.com/24457941/64475114-ddf85700-d1b0-11e9-8c70-8f117b9fc8db.png)

Reply:
FileInputStream close and ByteArrayInputStream close implements different, here not problem.